core/src/main/java/io/kestra/core/runners/ExecutionDelay.java

```diff
@@ -25,8 +24,24 @@ public class ExecutionDelay {
 
     @NotNull State.Type state;
 
+    @NotNull DelayType delayType;
+
     @JsonIgnore
     public String uid() {
         return String.join("_", executionId, taskRunId);
     }
+
+    /**
+     * For previous version, return RESUME_FLOW by default as it was the only case
+     * @return DelayType representing the action to do when
+     */
+    public DelayType getDelayType() {
+        return delayType == null ? DelayType.RESUME_FLOW : delayType;
+    }
+
+    public enum DelayType {
+        RESUME_FLOW,
```

Comment:
```suggestion
        RESUME_FLOW,
```

we already have retry, restart and replay, now you also add RESUME 😂 if this fits into replay or restart, perhaps worth changing to RESTART_FLOW or REPLAY_FLOW?

Reply:
I don't add resume, resume is the default behavior and the reason ExecutionDelay was created at first, and its for dev reason anyway